localparticipant.go

```diff
@@ -674,3 +678,140 @@ func (p *LocalParticipant) cleanup() {
 	p.rpcPendingAcks = &sync.Map{}
 	p.rpcPendingResponses = &sync.Map{}
 }
+
+func (p *LocalParticipant) StreamText(options StreamTextOptions) *TextStreamWriter {
+	if options.StreamId == nil {
+		streamId := uuid.New().String()
+		options.StreamId = &streamId
+	}
+
+	if options.Attributes == nil {
+		options.Attributes = make(map[string]string)
+	}
+
+	info := TextStreamInfo{
+		baseStreamInfo: &baseStreamInfo{
+			Id:       *options.StreamId,
+			MimeType: "text/plain",
+			Topic:    options.Topic,
+			// Q: Is this correct?
+			Timestamp:  time.Now().Unix(),
+			Size:       options.TotalSize,
+			Attributes: options.Attributes,
+		},
+	}
+
+	header := &livekit.DataStream_Header{
+		StreamId:    info.Id,
+		MimeType:    info.MimeType,
+		Topic:       info.Topic,
+		Timestamp:   info.Timestamp,
+		TotalLength: info.Size,
+		ContentHeader: &livekit.DataStream_Header_TextHeader{
+			TextHeader: &livekit.DataStream_TextHeader{
+				OperationType: livekit.DataStream_CREATE,
+			},
+		},
+	}
+	if options.ReplyToStreamId != nil {
+		if textHeader, ok := header.ContentHeader.(*livekit.DataStream_Header_TextHeader); ok {
+			textHeader.TextHeader.ReplyToStreamId = *options.ReplyToStreamId
+		}
+	}
+
+	writer := newTextStreamWriter(info, header, p.engine, options.DestinationIdentities, options.OnProgress)
+	p.engine.OnClose(writer.onClose)
+
+	return writer
+}
+
+func (p *LocalParticipant) SendText(text string, options StreamTextOptions) TextStreamInfo {
+	if options.TotalSize == nil {
```

Comment:
does this need to be a pointer? can it check for 0?

Reply:
yeah fair, will fix this

Reply:
Left it as a pointer as that's what the proto message expects, or should I change it to 0 for the public API and convert internally?

Reply:
not having a pointer makes it nicer to use, but that will one field different from protocol 🤔 . Easier decision if 0 is a valid value. If 0 is not valid, I think keeping it a value and convert to pointer internally would be better.

Reply:
Got it